### PR TITLE
Move printing to AtCore with plugin support

### PR DESCRIPTION
**Pull request by sithlord48**

### CODE REVIEW

File: src/atcore.cpp [hunk 1]
```diff
@@ -118,4 +121,39 @@ QList<QSerialPortInfo> AtCore::serialPorts() const
     return QList<QSerialPortInfo>();
 }
 
+void AtCore::newMessage(const QByteArray& msg)
+{
+    lastMessage = msg;
+    emit(receivedMessage(lastMessage));
+}
```
Comment (patrickelectric) [inline src/atcore.cpp:128]: Space between functions

File: src/plugins/grblplugin.cpp [hunk 2]
```diff
@@ -9,3 +9,7 @@ QString GrblPlugin::name() const
 GrblPlugin::GrblPlugin(QObject* parent)
 {
 }
```
Comment (patrickelectric) [inline src/plugins/grblplugin.cpp:12]: Space

File: src/plugins/grblplugin.cpp [hunk 3]
```diff
@@ -9,3 +9,7 @@ QString GrblPlugin::name() const
...
+bool GrblPlugin::readyForNextCommand(const QString& lastMessage)
+{
```
Comment (patrickelectric) [inline src/plugins/grblplugin.cpp:15]: Add qunused

File: src/plugins/repetierplugin.cpp [hunk 4]
```diff
@@ -9,3 +10,13 @@ QString RepetierPlugin::name() const
 RepetierPlugin::RepetierPlugin(QObject* parent)
 {
 }
+
+bool RepetierPlugin::readyForNextCommand(const QString& lastMessage)
+{
+    if (lastMessage.contains("ok") || lastMessage.contains("wait")){
+        return true;
+    }
+    else{
```
Comment (patrickelectric) [inline src/plugins/repetierplugin.cpp]: Else is not necessary

File: src/protocollayer.cpp [hunk 5]
```diff
@@ -1,9 +1,9 @@
 #include "protocollayer.h"
 
-#include <QFile>
-#include <QEventLoop>
-#include <QTextStream>
-#include <QDebug>
+//#include <QFile>
+//#include <QEventLoop>
+//#include <QTextStream>
+//#include <QDebug>
```
Comment (patrickelectric) [inline src/protocollayer.cpp]: Remove this comments

**Review by patrickelectric — CHANGES_REQUESTED** [review]

File: src/protocollayer.cpp (hunk 5 above)
Comment (sithlord48) [reply src/protocollayer.cpp]: :heavy_check_mark:

File: src/plugins/grblplugin.cpp (hunk 3 above)
Comment (sithlord48) [reply src/plugins/grblplugin.cpp:15]: :heavy_check_mark:  Q_UNUSED(lastMessage)

File: src/atcore.cpp (hunk 1 above)
Comment (sithlord48) [reply src/atcore.cpp:128]: :heavy_check_mark:

File: src/plugins/grblplugin.cpp (hunk 2 above)
Comment (sithlord48) [reply src/plugins/grblplugin.cpp:12]: :heavy_check_mark:

File: src/plugins/repetierplugin.cpp (hunk 4 above)
Comment (sithlord48) [reply src/plugins/repetierplugin.cpp]: :heavy_check_mark:

**Pull request closed without merging**